libcloud/dns/drivers/worldwidedns.py

```diff
@@ -0,0 +1,506 @@
+        :type     reseller_id: ``str``
+
+        :param    secure: Weither to use HTTPS or HTTP. Note: Some providers
+                only support HTTPS, and it is on by default.
+        :type     secure: ``bool``
+
+        :param    host: Override hostname used for connections.
+        :type     host: ``str``
+
+        :param    port: Override port used for connections.
+        :type     port: ``int``
+
+        :return: ``None``
+        """
+        super(WorldWideDNSDriver, self).__init__(key=key, secret=secret,
+                                                 secure=secure, host=host,
+                                                 port=port, **kwargs)
+        self.reseller_id = reseller_id
+
+    def list_zones(self):
+        """
+        Return a list of zones.
+
+        :return: ``list`` of :class:`Zone`
+
+        For more info, please see:
+        https://www.worldwidedns.net/dns_api_protocol_list.asp
+        or
+        https://www.worldwidedns.net/dns_api_protocol_list_reseller.asp
+        """
+        action = '/api_dns_list.asp'
+        if self.reseller_id is not None:
+            action = '/api_dns_list_reseller.asp'
+        zones = self.connection.request(action)
+        if len(zones.body) == 0:
+            return []
+        else:
+            return self._to_zones(zones.body)
+
+    def list_records(self, zone):
+        """
+        Return a list of records for the provided zone.
+
+        :param zone: Zone to list records for.
+        :type zone: :class:`Zone`
+
+        :return: ``list`` of :class:`Record`
+        """
+        records = self._to_records(zone)
+        return records
+
+    def iterate_records(self, zone):
+        """
+        Return a generator to iterate over records for the provided zone.
+
+        :param zone: Zone to list records for.
+        :type zone: :class:`Zone`
+
+        :rtype: ``generator`` of :class:`Record`
+        """
+        records = self._to_records(zone)
+        for record in records:
+            yield record
+
+    def get_zone(self, zone_id):
+        """
+        Return a Zone instance.
+
+        :param zone_id: ID of the required zone
+        :type  zone_id: ``str``
+
+        :rtype: :class:`Zone`
+        """
+        zones = self.list_zones()
+        zone = [zone for zone in zones if zone.id == zone_id]
+        if len(zone) == 0:
+            raise ZoneDoesNotExistError(driver=self,
+                                        value="The zone doesn't exists",
+                                        zone_id=zone_id)
+        return zone[0]
+
+    def get_record(self, zone_id, record_id):
+        """
+        Return a Record instance.
+
+        :param zone_id: ID of the required zone
+        :type  zone_id: ``str``
+
+        :param record_id: ID of the required record
+        :type  record_id: ``str``
+
+        :rtype: :class:`Record`
+        """
+        zone = self.get_zone(zone_id)
+        records = self.list_records(zone)
```

Comment:
I assume there isn't any more efficient way to retrieve a single record (e.g. filter by id or similar)?